### ISSUE TITLE
Better handling of roundabouts

### ISSUE THREAD

**Issue by hungerburg** [waymarkedtrails/waymarked-trails-site#389]
Yesterday I created a cycling route that passes by a roundabout. Now the elevation profile has a gap.

Currently, the route enters the roundabout, goes full circle and then jumps to the exit point.

Making the route only follow the actually used part of the roundabout would remedy the gap.

Seems like a nice pastime to ponder over a solution?

**Comment by Nekzuris** [comment]
I think the issues is your cycling route, it shouldn't have the whole roundabout in it, you should split the roundabout way and only add the used part of the roundabout in the route relation, that should solve your gap in elevation profile.

But the issues with the gap in elevation profile is still present with route that use one way twice (in both direction)

**Comment by ppete2** [comment]
No: splitting roundabouts just to get closed, linear relations is not good mapping practice. Many OSM mappers even refuse this kind of splitting. A roundabout is one common, single feature, it has the same topograhic logic like a punctual crossing. In OSM editors like JOSM's relation editor, a roundbout logically is a punctual feature and therefor not seperating a route: The route stays connected even if the roundabout is not splitted.

I support the feature request of better integration of unsplitted roundabouts into WMT's elevation profile/GPX-download. But i also understand that this is not a trivial task since WMT hast to create it's own split logic.

**Comment by lonvia** [comment]
At thi point, WMT doesn't support split routes per direction at all for the elevation profile and/or route download. Once it does, it will split roundabouts appropriately. No need to split them manually.

**Comment by lonvia** [comment]
That's solved with the new route profiles.

**Comment by ppete2** [comment]
I'm afraid handling of roundabouts within a route for download is not working as it should do. As an example take cycling route https://www.openstreetmap.org/relation/19726489
The route contains 3 unsplitted roundabouts. WMT seems do split these roundabouts correctly. But finally creates 6 individual tracks instead of 1 track of the whole route if downloading the route (format=GPX, forward).

**Comment by lonvia** [comment]
Looks alright to me at https://cycling.waymarkedtrails.org/#route?id=19726489&type=relation&map=12.0/48.1086/14.126 The analyser reports one route. If you refer to the GPX, then it is intentional that the route is split whenever it forks.

And on a general note: do not comment on closed issues. Open a new one if you have issues.
